### PR TITLE
Handle null pointers in deep-copy logic

### PR DESCRIPTION
**Pull request by andschwa**
This adds some (admittedly barebones) tests for null pointers, and then handles them.

### CODE REVIEW

File: tools/oeedger8r/Emitter.ml [hunk 1]
```diff
@@ -764,26 +778,33 @@ let gen_enclave_code (ec : enclave_content) (ep : edger8r_params) =
...
+                (arg ^ gen_c_deref param_count)
+                param_count setter)
+             (get_deepcopy_members (get_param_atype ptype))) ]
+      |> List.flatten )
   in
   let oe_gen_in_and_inout_setters (plist : pdecl list) =
     let params =
       flatten_map
-        (oe_gen_set_pointers "" "1" (fun p ->
+        (oe_gen_set_pointers "" "" "1" (fun p ->
```
Comment (andschwa) [inline tools/oeedger8r/Emitter.ml]: @anakrish I have an internal TODO to refactor arg + arg^ parameters here, but it'll come later.

**Comment by andschwa** [comment]
Part of #1839.

File: include/openenclave/edger8r/common.h [hunk 2]
```diff
@@ -81,7 +81,7 @@ OE_INLINE oe_result_t oe_add_size(size_t* total, size_t size)
  * buffer. Make sure that the buffer has enough space.
  */
 #define OE_SET_IN_POINTER(argname, argsize, argtype)                       \
-    if (pargs_in->argname)                                                 \
+    if (pargs_in->argname) /* TODO: This check is now extraneous. */       \
```
Comment (anakrish) [inline include/openenclave/edger8r/common.h]: This header, though internal, is installed and is user visible when debugging.
 I am not comfortable with introducing a new TODO in the last week of v0.6 development.

Comment (andschwa) [reply include/openenclave/edger8r/common.h]: @anakrish This is literally just noting that we do `if(pargs_in->argname)` in the generated code now. There's no harm in doing it here, but it's superfluous. If you want I can just remove the TODOs until if/when we decide to remove the extra checks.

File: tools/oeedger8r/Emitter.ml [hunk 3]
```diff
@@ -764,26 +778,33 @@ let gen_enclave_code (ec : enclave_content) (ep : edger8r_params) =
...
-         oe_get_param_count (ptype, decl, "pargs_in->" ^ prefix)
-       in
-       flatten_map
-         (oe_gen_set_pointers
-            (arg ^ gen_c_deref param_count)
-            param_count setter)
-         (get_deepcopy_members (get_param_atype ptype))) ]
-    |> List.flatten
+    let check_expr =
+      (* TODO: Also gross. *)
```
Comment (andschwa) [inline tools/oeedger8r/Emitter.ml]: @anakrish This is just a code-style comment of something I want to clean up eventually.

**Comment by andschwa** [comment]
@anakrish this PR is opened separately in order to keep PRs to particular changes, I have two more pending sets of commits to open on top of this. We'll do a final clean up before branching for release.

**Comment by andschwa** [comment]
@anakrish I removed the TODOs and replaced with NOTEs in the edger8r for potential cleanup.

File: tools/oeedger8r/Emitter.ml [hunk 4]
```diff
@@ -637,21 +643,25 @@ let gen_enclave_code (ec : enclave_content) (ep : edger8r_params) =
...
+          String.concat " && " (List.filter (( <> ) "") [parent; arg])
+        in
+        gen_c_for count
+          ( [ [sprintf "if (%s)" check_expr]
+            ; [sprintf "    OE_ADD_SIZE(%s, %s);" buffer size]
+            ; (let param_count =
+                 oe_get_param_count (ptype, decl, "_args." ^ prefix)
+               in
+               flatten_map
+                 (gen_add_size arg (arg ^ gen_c_deref param_count) param_count)
```
Comment (anakrish) [inline tools/oeedger8r/Emitter.ml]: Overall looks good. Can you double-check if we need to pass check_expr to the children?
Intuitively i would expect that check_expr gets passed down each level and each level adds another condition to the check_expr

Comment (andschwa) [reply tools/oeedger8r/Emitter.ml]: Yup we did, thank you! Fixed.

**Comment by andschwa** [comment]
We keep having CI issues.

bors r-

**Pull request merged**